gmcs/linglib/morphotactics.py

```diff
@@ -1047,6 +1053,48 @@ def write_pc_adj_syntactic_behavior(lrt, mylang, choices):
                 mylang.add(lrt.identifier() +
                            ''' := [ SYNSEM.LOCAL.CAT.HEAD.PRD - ].''')
 
+
+'''
+OZ-2020-09-24
+The interrogative inflection without lexical threading
+requires an interrogative supertype (which will work for
+grammars where the same morphology is used for polar and wh-questions),
+ and a further subtype which can be added if this morphology
+ is used only for polar questions. Note that ambiguity is possible
+ if both rules are added to the same PC with the same orthography,
+ but that is unlikely.
+ An alternative would be to add something like this INSTEAD of the user-specified rule:
+wh-subj-lex-rule := itrg-lex-rule &
+                [ SYNSEM.LOCAL.CAT.VAL.SUBJ < [ NON-LOCAL.QUE.LIST cons ] > ].
+wh-obj-lex-rule := itrg-lex-rule &
+                [ SYNSEM.LOCAL.CAT.VAL [ SUBJ < [ NON-LOCAL.QUE.LIST < > ] >,
+                COMPS < [ NON-LOCAL.QUE.LIST cons ] > ] ].
+-- but it is not clear to me how to properly incorporate this into the hierarchy,
+and at what stage.
+'''
+
+
+def write_interrogative_rules(lrt, mylang):
+    ITRG_LEX_RULE = '''itrg-lex-rule := add-only-no-ccont-rule & 
+    [ SYNSEM.LOCAL.CONT.HOOK.INDEX.SF ques ].'''
+    PROP_LEX_RULE = '''prop-lex-rule := add-only-no-ccont-rule & 
+    [ SYNSEM.LOCAL.CONT.HOOK.INDEX.SF prop ].'''
+    POLAR_LEX_RULE = '''polar-lex-rule := itrg-lex-rule & 
+    [ SYNSEM.LOCAL.CAT.VAL [ SUBJ < [ NON-LOCAL.QUE.LIST < > ] >, COMPS non-wh-list ] ].'''
```

Comment:
Why are these variables capitalized? Did you intend to make them constants somewhere?

Reply:
Hmm good question, I was thinking of them as constants because it is not like they are to be modified at any point? They are to be passed to tdl.add() as is, in various places in the code?